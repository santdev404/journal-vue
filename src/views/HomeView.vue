<template>

  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
  </div>

  <h1>Hello there!!</h1>
  <button @click="goToDayBook" class="btn btn-primary">primary</button>
  <button @click="goToDayBook" class="btn btn-secondary">secondary</button>
  <button @click="goToDayBook" class="btn btn-success">success</button>

</template>

<script>

export default {
  methods:{
    goToDayBook()
    {
      this.$router.push({  name: 'no-entry' })
    }
  }

}

</script>
