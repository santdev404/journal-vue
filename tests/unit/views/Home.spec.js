import { shallowMount } from "@vue/test-utils"
import HomeView from '@/views/HomeView'



describe('Prueba en el Home View', () =>{
    test('test de renderizar el componente correctamente', () =>{
        
        const wrapper = shallowMount(HomeView)
        expect( wrapper.html()).toMatchSnapshot()

    })


    test('Hace click en un boton debe redireccionar a no-entry', () =>{

        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(HomeView, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')

        expect( mockRouter.push ).toHaveBeenCalled()
        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'no-entry'})


    
    })

})