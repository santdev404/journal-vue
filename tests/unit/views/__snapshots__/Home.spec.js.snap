// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el Home View test de renderizar el componente correctamente 1`] = `
<div class="home"><img alt="Vue logo" src=""></div>
<h1>Hello there!!</h1>
<button class="btn btn-primary">primary</button>
<button class="btn btn-secondary">secondary</button>
<button class="btn btn-success">success</button>
`;
